src/typechecker/collect.lisp

```diff
@@ -1,226 +0,0 @@
-;;;;
-;;;; Generic traversals for collecting lists of items referenced from
-;;;; within a nested AST structure.
-;;;;
-
-(defpackage #:coalton-impl/parser/collect
```

Comment:
This file still thinks it's in the parser, and @eliaslfox verified it's fine to delete